temporal-sdk/src/main/java/io/temporal/client/UpdateWithStartWorkflowOperation.java

```diff
@@ -0,0 +1,415 @@
+   * @param arg5 fifth request function parameter
+   * @param arg6 sixth request function parameter
+   */
+  public static <A1, A2, A3, A4, A5, A6> Builder<Void> newBuilder(
+      Functions.Proc6<A1, A2, A3, A4, A5, A6> request,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5,
+      A6 arg6) {
+    return new Builder<>(() -> request.apply(arg1, arg2, arg3, arg4, arg5, arg6));
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation}.
+   *
+   * @param updateName name of the update handler
+   * @param resultClass class of the update return value
+   * @param args update request arguments
+   */
+  public static <R> Builder<R> newBuilder(String updateName, Class<R> resultClass, Object[] args) {
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setResultClass(resultClass).setUpdateName(updateName), args);
+  }
+
+  private WorkflowStub stub;
+
+  private UpdateOptions<R> options;
+
+  private Object[] updateArgs;
+
+  private final CompletableFuture<WorkflowUpdateHandle<R>> handle;
+
+  private final Functions.Proc request;
+
+  private UpdateWithStartWorkflowOperation(
+      UpdateOptions<R> options, Functions.Proc request, Object[] updateArgs) {
+    this.options = options;
+    this.updateArgs = updateArgs;
+    this.handle = new CompletableFuture<>();
+    this.request = request;
+  }
+
+  WorkflowUpdateHandle<R> invoke(Functions.Proc workflow) {
+    WorkflowInvocationHandler.initAsyncInvocation(
+        WorkflowInvocationHandler.InvocationType.UPDATE_WITH_START, this);
+    try {
+      request.apply();
+      workflow.apply();
+      stub.updateWithStart(this, this.updateArgs);
+      return this.handle.get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      Throwable cause = e.getCause();
+      throw (cause instanceof RuntimeException
+          ? (RuntimeException) cause
+          : new RuntimeException(cause));
+    } finally {
+      WorkflowInvocationHandler.closeAsyncInvocation();
+    }
+  }
+
+  void prepareUpdate(
+      WorkflowStub stub, String updateName, Class resultClass, Type resultType, Object[] args) {
+    setStub(stub);
+    this.updateArgs = args;
+    this.options =
+        this.options.toBuilder()
+            .setUpdateName(updateName)
+            .setResultClass(resultClass)
+            .setResultType(resultType)
+            .build();
+  }
+
+  void prepareStart(WorkflowStub stub) {
+    setStub(stub);
+  }
+
+  /** Returns the result of the update request. */
+  public R getResult() throws ExecutionException, InterruptedException {
+    return this.getUpdateHandle().get().getResultAsync().get();
+  }
+
+  public Future<WorkflowUpdateHandle<R>> getUpdateHandle() {
+    return this.handle;
+  }
+
+  public void setUpdateHandle(WorkflowUpdateHandle<R> updateHandle) {
```

Comment:
This should not be public

Reply:
I see your calling it from the root interceptor and that is probably the root of the issue. I think this should be set at the top of the interceptor chain and _I think_ that will also fix the visibility here

Reply:
Yeah, I had called that out before (in the first version of the PR); that's a good idea on how to solve that. https://github.com/temporalio/sdk-java/pull/2199/commits/c2e0b4237bfe1a87d2ef20bbc0c8b5401e18c096